### PR TITLE
[Kotlin Migration] TagsArrayAdapter

### PR DESCRIPTION
**Pull request by Akshay0701**

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/dialogs/tags/TagsArrayAdapter.kt [hunk 1]
```diff
@@ -13,160 +13,121 @@
...
+import androidx.annotation.VisibleForTesting
+import androidx.recyclerview.widget.RecyclerView
+import com.ichi2.anki.R
+import com.ichi2.ui.CheckBoxTriStates
+import com.ichi2.utils.FilterResultsUtils
+import java.util.*
+import kotlin.collections.ArrayList
+
+/**
+ * A reference to the {@link TagsList} passed.
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/dialogs/tags/TagsArrayAdapter.kt:33]: Can this be changed to a `@param` or similar - it's not a doc for the class

**Review by david-allison-1 — CHANGES_REQUESTED** [review]
One Q

File: AnkiDroid/src/main/java/com/ichi2/anki/dialogs/tags/TagsArrayAdapter.kt (hunk 1 above)
Comment (Akshay0701) [reply AnkiDroid/src/main/java/com/ichi2/anki/dialogs/tags/TagsArrayAdapter.kt]: done

**Review by david-allison-1 — APPROVED** [review]
LGTM

**Review by mikehardy — APPROVED** [review]

**Pull request merged**